### DISCUSSION TITLE
QUIC

### DISCUSSION THREAD

**Discussion by emiltin** [rsmp-nordic/rsmp_core_v4#2] [Ideas]
QUIC is a new alternative to TCP. From https://www.emqx.com/en/blog/quic-protocol-the-features-use-cases-and-impact-for-iot-iov:

QUIC (Quick UDP Internet Connections) is a protocol Google has developed to improve the speed and reliability of web connections. It is designed to replace the Transmission Control Protocol (TCP) used in the current Internet infrastructure. QUIC is built on top of the User Datagram Protocol (UDP).

QUIC uses a combination of encryption and multiplexing to provide improved security and faster data transfer. It allows multiple streams of data to be sent over a single connection, reducing latency and improving throughput. QUIC also includes features such as congestion control and flow control to manage network congestion and ensure smooth data transmission.

The Internet Engineering Task Force (IETF) is standardizing QUIC, and major web browsers and servers are adopting it. QUIC has been shown to improve web page loading times and reduce the occurrence of disconnections compared to TCP, especially in high-latency and spotty networks such as mobile networks.

There are also some substantial benefits of running MQTT over QUIC.

**Comment by emiltin** [comment]
QUIC uses end-to-end encryption. There is no clear-text version of the protocol so to negotiate a QUIC connection means doing cryptography and security with TLS 1.3

**Comment by emiltin** [comment]
In Oct. 2020, Facebook announced that “more than 75% of [their] Internet traffic uses QUIC and HTTP/3”
https://engineering.fb.com/2020/10/21/networking-traffic/how-facebook-isbringing-quic-to-billions/

Around 7-8% of all websites use QUIC (2023):
https://w3techs.com/technologies/details/ce-quic

QUIC use is highest in Denmark, Sweden, Norway, and Switzerland (2022):
https://blog.apnic.net/2022/07/11/a-look-at-quic-use/

**Comment by otterdahl** [comment]
QUIC seems like an intressting option. Hopefully it can address issues regarding congestion control which [we've experienced](https://github.com/rsmp-nordic/rsmp_core/issues/50) with RSMP.

**Reply by emiltin** [reply]
I think it's the future, but QUIC support in MQTT is relatively new and I don't think prioritizing certain topics is supported yet in eg EMQT, but will arrive at some point. But what we can now already is split huge data blobs into smaller messages, so live update has a chance to come through.
